### PR TITLE
Added the change for props change rerender Issue

### PR DESCRIPTION
**Pull request by ashishkumbhare116**
### Summary

**What was changed:**
-  Fixed State is not re-rendering when we update icon prop.

**Why it was changed:**
- Re-render was not happening when we update icon prop.

### Testing

This change was tested using:

- [ ] WDIO
- [ ] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [ ] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-10244 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

**Review by saket2403 — APPROVED** [review]

File: packages/terra-tabs/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+* Changed
+  * Added the fix for changing the props not rerendering component.
```
Comment (sugan2416) [inline packages/terra-tabs/CHANGELOG.md:6]:
```suggestion
  * Fixes component re-render with updated icon.
```

File: packages/terra-tabs/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+* Changed
```
Comment (sugan2416) [inline packages/terra-tabs/CHANGELOG.md:5]:
```suggestion
* Fixed
```

File: packages/terra-tabs/CHANGELOG.md (hunk 1 above)
Comment (PK106552) [reply packages/terra-tabs/CHANGELOG.md]: Updated

File: packages/terra-tabs/CHANGELOG.md (hunk 2 above)
Comment (PK106552) [reply packages/terra-tabs/CHANGELOG.md]: Updated

**Review by sugan2416 — APPROVED** [review]

**Pull request merged**